packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -152,14 +155,79 @@ function buildAndRun(args: Flags, androidProject: AndroidProject) {
   }
 
   const adbPath = getAdbPath();
+
+  if (args.device === true) {
+    const device = await promptWhichDeviceToRun(adbPath);
+
+    if (device) {
+      if (device.type === 'emulator') {
+        return runOnEmulator(
+          args,
+          cmd,
+          packageName,
+          adbPath,
+          androidProject,
+          device.name,
+        );
+      }
+
+      if (device.type === 'device') {
+        return runOnSpecificDevice(
+          {
+            ...args,
+            deviceId: device.deviceId,
+          },
+          cmd,
+          packageName,
+          adbPath,
+          androidProject,
+        );
+      }
+    }
+  }
+
   if (args.deviceId) {
     return runOnSpecificDevice(args, cmd, packageName, adbPath, androidProject);
-  } else {
-    return runOnAllDevices(args, cmd, packageName, adbPath, androidProject);
   }
+
+  return runOnEmulator(args, cmd, packageName, adbPath, androidProject);
 }
 
-function runOnSpecificDevice(
+async function promptWhichDeviceToRun(adbPath: string) {
+  const devices = adb.getAllAvailableDevices(adbPath);
```

Comment:
The iOS function is pure - doesn't ask for devices, receives an array instead. Would it make sense to move this outside of the function - might be useful to refactor the code in a follow up PR.

It looks like we're calling `adb.getAllAvailableDevices(adbPath);` and `adb.getDevices(adbPath)` definitely too many times (as opposed to just one on the iOS side of things).